docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,133 @@
+
+You can change the value in `config/initializers/react_on_rails` by updating as follows:
+
+```rb
+config.auto_load_bundle = true
+```
+
+### Update `.gitignore` file
+React on Rails, per the settings in your `config/webpacker.yml`, automatically generates pack files for components to be registered in the `packs/generated` directory. To avoid committing generated files into the version control system, Please update `.gitignore` to have 
+
+```gitignore
+### Generated React on Rails packs
+packs/generated
+```
+
+*Note, the directory might be different depending on the `source_entry_path` in `config/webpacker.yml`.*
+
+## Usage
+
+### Basic usage
+
+if the `webpacker.yml` file is configured as instructed [here](https://github.com/shakacode/shakapacker#configuration-and-code), With following configurations
+
+```yml
+default: &default
+    source_path: app/javascript
+    source_entry_path: packs 
+    public_root_path: public
+    public_output_path: packs
+    nested_entries: true
+# And more 
+```
+
+the directory structure will look like this
+```
+app/javascript:
+  └── packs:               # sets up webpack entries
+  │   └── application.js   # references ../src/my_component.js
+  │   └── application.css
+  └── src:                 # any directory name is fine. Referenced files need to be under source_path
+  │   └── component.js
+  └── stylesheets:
+  │   └── my_styles.css
+  └── images:
+      └── logo.svg
+public/packs                # webpack output
+```
+
+Now, To automatically register, `A_ComponentOne`, `B_ComponentOne`, `B_ComponentTwo` for the usage with `react_component` and `react_component_hash` helpers, Create directory structure as mentioned below:
+
+```
+app/javascript:
+  └── packs:                   
+  │   └── application.js     
+  │   └── application.css
+  └── src:                   
+  │   └── A
+  │   │ └── ...
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   └── A_ComponentOne.jsx
+  │   └── B
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   │ └── B_ComponentOne.jsx
+  │   │   │ └── B_ComponentTwo.jsx
+public/packs                
+```
+
+To register react component, It is no longer needed to create a pack entry and manually register it by calling `ReactOnRails.register`. With automatically generated packs, you can directly use `A_ComponentOne`, `B_ComponentOne` and `B_ComponentTwo` in rails view using:
+
+```erb
+    <%= react_component("A_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentTwo", {}, auto_load_bundle: true) %>    
+```
+
+If `A_ComponentOne` uses multiple HTML strings for server rendering, `react_component_hash` view helper can be used on rails view as illustrated below. Read more about [`react_component_hash` here](https://www.shakacode.com/react-on-rails/docs/api/view-helpers-api/#react_component_hash).
```

Comment:
I think you can remove the sentence and just make the name a link where it's first mentioned (and for `react_component` as well).

Reply:
Not resolved.